From f1e8371c3a52a591898b456d2ef2bbcfc14da6aa Mon Sep 17 00:00:00 2001
From: T H <socialentry@gmail.com>
Date: Mon, 16 Jul 2018 00:34:36 -0400
Subject: [PATCH 3/3] Snap: Fix saving HSR

* Add config to fix saving HSR on klte
* Add Denoise to Video menu

Change-Id: Ie8fb27ea52039890dc23c73eb4b4b2837341f893
---
 res/values/config.xml                   | 3 +++
 res/xml/video_preferences.xml           | 7 +++++++
 src/com/android/camera/VideoMenu.java   | 2 ++
 src/com/android/camera/VideoModule.java | 7 +++++++
 4 files changed, 19 insertions(+)

diff --git a/res/values/config.xml b/res/values/config.xml
index 9b506b072..9740002cb 100644
--- a/res/values/config.xml
+++ b/res/values/config.xml
@@ -73,4 +73,7 @@
 
     <!-- Minimum picture size (in pixel) to include in selection list -->
     <integer name="minimum_picture_size">900000</integer>
+
+    <!-- Use profile for HSR recording instead -->
+    <bool name="use_profile_for_hsr">false</bool>
 </resources>
diff --git a/res/xml/video_preferences.xml b/res/xml/video_preferences.xml
index e60dbc32e..8570eb25c 100644
--- a/res/xml/video_preferences.xml
+++ b/res/xml/video_preferences.xml
@@ -227,6 +227,13 @@
             camera:singleIcon="@drawable/ic_settings_powershutter"
             camera:entries="@array/pref_switch_entries"
             camera:entryValues="@array/pref_switch_entryvalues" />
+    <IconListPreference
+            camera:key="pref_camera_denoise_key"
+            camera:defaultValue="@string/pref_camera_denoise_default"
+            camera:title="@string/pref_camera_denoise_title"
+            camera:singleIcon="@drawable/ic_settings_denoise"
+            camera:entries="@array/pref_camera_denoise_entries"
+            camera:entryValues="@array/pref_camera_denoise_entryvalues" />
     <IconListPreference
             camera:key="pref_max_brightness"
             camera:defaultValue="@string/pref_camera_max_brightness_default"
diff --git a/src/com/android/camera/VideoMenu.java b/src/com/android/camera/VideoMenu.java
index 1de936eb9..d2e1aeace 100644
--- a/src/com/android/camera/VideoMenu.java
+++ b/src/com/android/camera/VideoMenu.java
@@ -126,6 +126,7 @@ public class VideoMenu extends MenuController
                 CameraSettings.KEY_VIDEO_HIGH_FRAME_RATE,
                 CameraSettings.KEY_DIS,
                 CameraSettings.KEY_POWER_SHUTTER,
+                CameraSettings.KEY_DENOISE,
                 CameraSettings.KEY_MAX_BRIGHTNESS
         };
         mOtherKeys2 = new String[] {
@@ -141,6 +142,7 @@ public class VideoMenu extends MenuController
                 CameraSettings.KEY_VIDEOCAMERA_FOCUS_TIME,
                 CameraSettings.KEY_VIDEO_HIGH_FRAME_RATE,
                 CameraSettings.KEY_POWER_SHUTTER,
+                CameraSettings.KEY_DENOISE,
                 CameraSettings.KEY_MAX_BRIGHTNESS,
                 CameraSettings.KEY_SEE_MORE,
                 CameraSettings.KEY_NOISE_REDUCTION,
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 14732483c..f0c787ebd 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -1764,6 +1764,13 @@ public class VideoModule implements CameraModule,
             !mCaptureTimeLapse && !isHFR) {
             mProfile.fileFormat = MediaRecorder.OutputFormat.THREE_GPP;
         }
+
+        // On klte, captureRate > 0 fails saving HSR, it works just setting the profile
+        boolean useProfileForHSR = mActivity.getResources().getBoolean(R.bool.use_profile_for_hsr);
+        if (isHSR && useProfileForHSR) {
+            captureRate = 0;
+        }
+
         // Set params individually for HFR case, as we do not want to encode audio
         if ((isHFR || isHSR) && captureRate > 0) {
             if (isHSR) {
-- 
2.17.1

